### PR TITLE
[zh-cn]Locatlization blog:2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md

### PR DESCRIPTION
**Pull request by my-git9**
content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md

### CODE REVIEW

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
+---
+layout: blog
+title: "Kubernetes 1.30: Preventing unauthorized volume mode conversion moves to GA"
+date: 2024-04-30
+slug: prevent-unauthorized-volume-mode-conversion-ga
+---
+
+<!--
+layout: blog
+title: "Kubernetes 1.30：防止未经授权的卷模式转换进阶到 GA"
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:10]: This should be line 3.

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+**译者:** Xin Li (DaoCloud)
+
+<!--
+With the release of Kubernetes 1.30, the feature to prevent the modification of the volume mode
+of a [PersistentVolumeClaim](/docs/concepts/storage/persistent-volumes/) that was created from
+an existing VolumeSnapshot in a Kubernetes cluster, has moved to GA!
+-->
+随着 Kubernetes 1.30 的发布，防止修改从 Kubernetes 集群中现有
+VolumeSnapshot 创建的 [PersistentVolumeClaim](/zh-cn/docs/concepts/storage/persistent-volumes/)
+的卷模式的特性已移至 GA！
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:28]:
```suggestion
的卷模式的特性已被升级至 GA！
```

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+to create a PersistentVolumeClaim (shortened as PVC) from an existing VolumeSnapshot in
+the Kubernetes cluster. The PVC spec includes a dataSource field, which can point to an
+existing VolumeSnapshot instance.
+Visit [Create a PersistentVolumeClaim from a Volume Snapshot](/docs/concepts/storage/persistent-volumes/#create-persistent-volume-claim-from-volume-snapshot) 
+for more details on how to create a PVC from an existing VolumeSnapshot in a Kubernetes cluster.
+-->
+## 问题
+
+PersistentVolumeClaim 的[卷模式](/zh-cn/docs/concepts/storage/persistent-volumes/#volume-mode)
+是指存储设备上的底层卷是否被格式化为文件系统或作为原始块设备呈现给使用它的 Pod。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:47]:
```suggestion
是指存储设备上的底层卷是被格式化为某文件系统还是作为原始块设备呈现给使用它的 Pod。
```

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+用户可以利用自 Kubernetes v1.20 以来一直稳定的 VolumeSnapshot 特性，从
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:49]:
```suggestion
用户可以利用自 Kubernetes v1.20 以来一直稳定的 VolumeSnapshot 特性，基于
```

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+Kubernetes 集群中现有的 VolumeSnapshot 创建 PersistentVolumeClaim（简称 PVC）。
+PVC 规范包括一个 dataSource 字段，它可以指向现有的 VolumeSnapshot 实例。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:51]:
```suggestion
PVC 规约中包括一个 `dataSource` 字段，它可以指向现有的 VolumeSnapshot 实例。
```

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+有关如何从 Kubernetes 集群中现有 VolumeSnapshot 创建 PVC 的更多详细信息，
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:52]:
```suggestion
有关如何基于 Kubernetes 集群中现有 VolumeSnapshot 创建 PVC 的更多详细信息，
```

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+请访问[从卷快照创建 PersistentVolumeClaim](/zh-cn/docs/concepts/storage/persistent-volumes/#create-persistent-volume-claim-from-volume-snapshot)。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:53]:
```suggestion
请访问[使用卷快照创建 PersistentVolumeClaim](/zh-cn/docs/concepts/storage/persistent-volumes/#create-persistent-volume-claim-from-volume-snapshot)。
```

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+
+This presents a security gap that allows malicious users to potentially exploit an
+as-yet-unknown vulnerability in the host operating system.
+
+There is a valid use case to allow some users to perform such conversions. Typically, storage backup
+vendors convert the volume mode during the course of a backup operation, to retrieve changed blocks 
+for greater efficiency of operations. This prevents Kubernetes from blocking the operation completely
+and presents a challenge in distinguishing trusted users from malicious ones.
+-->
+当利用上述特性时，没有逻辑来验证拍摄快照的原始卷的模式是否与新创建的卷的模式匹配。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:67]:
```suggestion
当利用上述特性时，没有逻辑来验证制作快照的原始卷的模式是否与新创建的卷的模式匹配。
```

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+
+这带来了一个安全漏洞，允许恶意用户潜在地利用主机操作系统中未知的漏洞。
+
+有一个有效的用例允许某些用户执行此类转换。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:71]:
```suggestion
有一个合法的场景允许某些用户执行此类转换。
```

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 10]
```diff
@@ -0,0 +1,197 @@
...
+通常，存储备份供应商会在备份操作过程中转换卷模式，以检索更改的块以提高操作效率。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:72]:
```suggestion
通常，存储备份供应商会在备份操作过程中转换卷模式，通过检索已被更改的块来提高操作效率。
```

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+这使得 Kubernetes 无法完全阻止操作，并且给区分可信用户和恶意用户带来了挑战。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:73]:
```suggestion
这使得 Kubernetes 无法完全阻止此类操作，但给区分可信用户和恶意用户带来了挑战。
```

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+It is up to the cluster administrator to provide these rights only to trusted users
+or applications, like backup vendors.
+Users apart from such authorized ones will never be allowed to modify the volume mode
+of a PVC when it is being created from a VolumeSnapshot.
+-->
+## 防止未经授权的用户转换卷模式
+
+在此上下文中，授权用户是有权对 VolumeSnapshotContents（集群级资源）执行
+**update** 或 **patch** 操作的用户。
+集群管理员应仅向受信任的用户或应用程序（例如备份供应商）提供这些权限。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:89]:
```suggestion
集群管理员应仅向受信任的用户或应用程序（例如备份供应商）赋予这些权限。
```

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 13]
```diff
@@ -0,0 +1,197 @@
...
+2. Identify the VolumeSnapshotContent bound to the above VolumeSnapshot.
+-->
+要转换卷模式，授权用户必须执行以下操作：
+
+1. 标识要用作给定命名空间中新创建的 PVC 的数据源的 VolumeSnapshot。
+2. 识别与上述 VolumeSnapshot 绑定的 VolumeSnapshotContent。
+
+  ```shell
+  kubectl describe volumesnapshot -n <namespace> <name>
+  ```
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:106]:
````suggestion
   ```shell
   kubectl describe volumesnapshot -n <namespace> <name>
   ```
````

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 14]
```diff
@@ -0,0 +1,197 @@
...
+3. 在 VolumeSnapshotContent 上添加 [`snapshot.storage.kubernetes.io/allow-volume-mode-change: "true"`](/zh-cn/docs/reference/labels-annotations-taints/#snapshot-storage-kubernetes-io-allowvolumemodechange)
+   注解，VolumeSnapshotContent 注解必须包含类似于以下清单片段的注解：
+
+  ```yaml
+  kind: VolumeSnapshotContent
+  metadata:
+    annotations:
+      - snapshot.storage.kubernetes.io/allow-volume-mode-change: "true"
+  ...
+  ```
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:121]:
````suggestion
   ```yaml
   kind: VolumeSnapshotContent
   metadata:
     annotations:
       - snapshot.storage.kubernetes.io/allow-volume-mode-change: "true"
  ...
  ```
````

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 15]
```diff
@@ -0,0 +1,197 @@
...
+  ```shell
+  kubectl describe volumesnapshot -n <namespace> <name>
+  ```
+
+<!--
+3. Add the annotation [`snapshot.storage.kubernetes.io/allow-volume-mode-change: "true"`](/docs/reference/labels-annotations-taints/#snapshot-storage-kubernetes-io-allowvolumemodechange)
+   to the above VolumeSnapshotContent. The VolumeSnapshotContent annotations must include one similar to the following manifest fragment:
+-->
+3. 在 VolumeSnapshotContent 上添加 [`snapshot.storage.kubernetes.io/allow-volume-mode-change: "true"`](/zh-cn/docs/reference/labels-annotations-taints/#snapshot-storage-kubernetes-io-allowvolumemodechange)
+   注解，VolumeSnapshotContent 注解必须包含类似于以下清单片段的注解：
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:113]:
```suggestion
   注解，VolumeSnapshotContent 注解必须包含类似于以下清单片段：
```

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 16]
```diff
@@ -0,0 +1,197 @@
...
+<!--
+**Note**: For pre-provisioned VolumeSnapshotContents, you must take an extra
+step of setting `spec.sourceVolumeMode` field to either `Filesystem` or `Block`,
+depending on the mode of the volume from which this snapshot was taken.
+
+An example is shown below:
+-->
+**注意**：对于预配置的 VolumeSnapshotContents，你必须执行额外的步骤，将
+`spec.sourceVolumeMode` 字段设置为 `Filesystem` 或 `Block`，
+具体取决于从中获取此快照的卷的模式。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:132]:
```suggestion
具体取决于用来制作此快照的卷的模式。
```

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 17]
```diff
@@ -0,0 +1,197 @@
...
+The `prevent-volume-mode-conversion` feature flag is enabled by default in the 
+external-provisioner `v4.0.0` and external-snapshotter `v7.0.0`. Volume mode change
+will be rejected when creating a PVC from a VolumeSnapshot unless the steps
+described above have been performed.
+-->
+## 需要采取的行动
+
+默认情况下，在 external-provisioner `v4.0.0` 和 external-snapshotter `v7.0.0`
+中启用 `prevent-volume-mode-conversion` 特性标志。
+从 VolumeSnapshot 创建 PVC 时，卷模式更改将被拒绝，除非已执行上述步骤。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:182]:
```suggestion
基于 VolumeSnapshot 来创建 PVC 时，卷模式更改将被拒绝，除非已执行上述步骤。
```

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 18]
```diff
@@ -0,0 +1,197 @@
...
+To determine which CSI external sidecar versions support this feature, please head
+over to the [CSI docs page](https://kubernetes-csi.github.io/docs/).
+For any queries or issues, join [Kubernetes on Slack](https://slack.k8s.io/) and
+create a thread in the #csi or #sig-storage channel. Alternately, create an issue in the
+CSI external-snapshotter [repository](https://github.com/kubernetes-csi/external-snapshotter).
+-->
+## 接下来
+
+要确定哪些 CSI 外部 sidecar 版本支持此功能，请前往 [CSI 文档页面](https://kubernetes-csi.github.io/docs/)。
+对于任何疑问或问题，请加入 [Slack 上的 Kubernetes](https://slack.k8s.io/) 并在 #csi 或 #sig-storage 通道中创建话题。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:196]:
```suggestion
对于任何疑问或问题，请加入 [Slack 上的 Kubernetes](https://slack.k8s.io/) 并在 #csi 或 #sig-storage 频道中发起讨论。
```

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md [hunk 19]
```diff
@@ -0,0 +1,197 @@
...
+或者，在 CSI 外部快照[存储库](https://github.com/kubernetes-csi/external-snapshotter)中创建问题。
```
Comment (tengqm) [inline content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:197]:
```suggestion
或者，在 CSI 外部快照[仓库](https://github.com/kubernetes-csi/external-snapshotter)中登记问题。
```

File: content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md (hunk 1 above)
Comment (my-git9) [reply content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:9]: Thank you for your review, but I don't understand what you mean here

Comment (tengqm) [reply content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md:9]: The translated `title` should not be the one in comment.
We are supposed to translate line 3.

Comment (my-git9) [reply content/zh-cn/blog/_posts/2024-02-21-prevent-unauthorized-volume-mode-conversion-ga.md]: Updated, thanks

**Pull request merged**